### PR TITLE
11977 Fixed proprietor tax id (business number)

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* /bcgov/entity#11977

*Description of changes:*
- deleted obsolete officer incorp number
- deleted obsolete role type Subscriber
- added Party and ApiPerson interfaces
- build/parse filing using correct schema
- converted party <-> orgPerson
- fixed lint issues
- app version = 3.5.21

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/common/ListPeopleAndRoles.vue [hunk 1]
```diff
@@ -48,11 +48,6 @@
                   <p class="email">{{ officerEmail(orgPerson) }}</p>
                 </div>
 
-                <div v-if="officerIncorpNumber(orgPerson)" class="mt-2 incorporation-number">
```
Comment (severinbeauvais) [inline src/components/common/ListPeopleAndRoles.vue:51]: Not used at the moment. Easy to add back in later if needed.

File: src/enums/roleTypes.ts [hunk 2]
```diff
@@ -1,10 +1,9 @@
 export enum RoleTypes {
-  DIRECTOR = 'Director',
   COMPLETING_PARTY = 'Completing Party',
   CUSTODIAN = 'Custodian',
+  DIRECTOR = 'Director',
   INCORPORATOR = 'Incorporator',
   LIQUIDATOR = 'Liquidator',
   PARTNER = 'Partner',
   PROPRIETOR = 'Proprietor',
-  SUBSCRIBER = 'Subscriber',
```
Comment (severinbeauvais) [inline src/enums/roleTypes.ts:9]: Not used in this UI and did not match [party schema](https://github.com/bcgov/business-schemas/blob/7be2b1722e015202d65c55e838daff2f132b3625/src/registry_schemas/schemas/party.json#L74), so I deleted this.

File: src/interfaces/filing-interfaces/filing-interfaces.ts [hunk 3]
```diff
@@ -41,7 +41,7 @@ export interface IncorporationFilingIF {
       phone: string
       extension?: number
     }
-    parties: OrgPersonIF[]
+    parties: PartyIF[]
```
Comment (severinbeauvais) [inline src/interfaces/filing-interfaces/filing-interfaces.ts:44]: OrgPersonIF is close enough to PartyIF that IA filings and dissolution filings previously worked, but the [incorporation application](https://github.com/bcgov/business-schemas/blob/main/src/registry_schemas/schemas/incorporation_application.json) and [dissolution](https://github.com/bcgov/business-schemas/blob/main/src/registry_schemas/schemas/dissolution.json) schemas do, in fact, specify the Party schema/interface.

TLDR: this change doesn't break IAs or dissolutions and is now correct.

File: src/interfaces/filing-interfaces/filing-interfaces.ts [hunk 4]
```diff
@@ -103,7 +103,7 @@ export interface RegistrationFilingIF {
       phone: string
     }
     nameRequest: RegistrationNameRequestIF
-    parties: OrgPersonIF[]
+    parties: PartyIF[]
```
Comment (severinbeauvais) [inline src/interfaces/filing-interfaces/filing-interfaces.ts:106]: Here the IF matters because PartyIF names a field "taxId" while OrgPersonIF names it "businessNumber".

File: src/mixins/filing-template-mixin.ts [hunk 5]
```diff
@@ -365,21 +365,12 @@ export default class FilingTemplateMixin extends Mixins(DateMixin) {
           legalType: this.getEntityType,
           nrNumber: this.getNameRequest.nrNumber
         },
-        parties: this.getAddPeopleAndRoleStep.orgPeople,
+        parties: this.orgPersonsToParties(this.getAddPeopleAndRoleStep.orgPeople),
```
Comment (severinbeauvais) [inline src/mixins/filing-template-mixin.ts:368]: The type conversion is done here, as close as possible to the interface between the UI and the API (since the UI calls it one thing and the API calls it another).

Ditto for line 438 below.

File: tests/unit/BusinessTypeConfirm.spec.ts [hunk 6]
```diff
@@ -13,16 +13,16 @@ describe('Business Type Confirm component', () => {
   it('works as expected when there initial business type confirm is false', async () => {
     wrapper = wrapperFactory(
       BusinessTypeConfirm,
-      { 
+      {
```
Comment (severinbeauvais) [inline tests/unit/BusinessTypeConfirm.spec.ts:16]: When I ran `npm run lint`, these issues were fixed automatically.

To all devs: please lint locally before committing code.

File: tests/unit/filing-template-mixin.spec.ts [hunk 7]
```diff
@@ -193,7 +193,7 @@ describe('Registration Filing', () => {
         naicsCode: '12345',
         naicsDescription: 'Some NAICS Description'
       },
-      businessTypeConfirm: false,
+      businessTypeConfirm: false
```
Comment (severinbeauvais) [inline tests/unit/filing-template-mixin.spec.ts:196]: Another lint fix.

(It's weird that lint allows a trailing comma for enums but not objects but oh well.)

**Review by argush3 — APPROVED** [review]

**Review by Lcyc29 — APPROVED** [review]
